core/src/main/java/io/opencensus/stats/StatsManager.java

```diff
@@ -32,5 +32,5 @@
   /**
    * Returns the default {@link StatsContextFactory}.
    */
-  abstract StatsContextFactory getStatsContextFactory();
+  public abstract StatsContextFactory getStatsContextFactory();
```

Comment:
> I understand in production recording and querying must share state. I think Stats.java is taking care of that. For the sake of unit testing, recording and querying do need to be separately injected, because one application component only deal with one of them.

That makes sense.  We just need to split the classes so that client code can pass around only the part that it needs.

> That would work. Since you will have StatsRecorder(), will you also deprecate the record() on StatsContext?

Yes, in the new design, StatsContext is a simple value class, and it doesn't have a `record` method.